docs/en_US/gsoc-ideas.md

```diff
@@ -0,0 +1,64 @@
+# GSoC 2024 Ideas List
+
+Hi there! This is the ideas list of OceanBase for Google Summer of Code 2024! 
+
+As a first-year mentor organization, we focus the development ideas on ob-operator, a kubernetes operator which helps deploy and manage OceanBase cluster on kubernetes cluster seamlessly. There are four projects for contributors, and we'll offer delicate guidance for every choice.
+
+Enjoy your summer of code!
+
+## 1. CLI tool
+
+* Project Description: The primary method to manage CRDs of ob-operator is manipulating YAML manifests which is not easy enough for new-coming users to get started. Within the CLI tool, features like component installation, demo setup,  resource management and necessary validation can be implemented.
+* Required Skills: Golang, Kubernetes
+* Project Size: medium/large
+* Expected Outcomes: A complete CLI tool to control and manage CRDs in ob-operator, in other words, to manage clusters, tenants, backups on ob-operator in kubernetes cluster. Or a module in the CLI tool.
+* References: 
+  * [kubernetes/client-go](https://github.com/kubernetes/client-go)
+  * [spf13/cobra](https://github.com/spf13/cobra)
+  * [manifoldco/promptui](https://github.com/manifoldco/promptui)
+
+
+## 2. Light-weighted Operations
+
+* Project Description: Use Light-weighted operation task types to influence the status of resources instead of modifying specifications of those bigger resources.
+* Required Skills: Golang, Development of Kubernetes controller, Docker
+* Project Size: medium
+* Expected Outcomes: Implementation of CRD(s) and corresponding controller(s) for cluster and tenant, which can trigger small operational actions and reveal the progress through `status` fields.
```

Comment:
1. "for cluster and tenant" need to be changed to "for obcluster and obtenant operations"
2. need to support all the operations ob-operator provides currently